macros/utils/table_operations/merge_sql.sql

```diff
@@ -1,14 +1,21 @@
-{% macro merge_sql(target_relation, tmp_relation, unique_key, dest_columns) -%}
-    {{ return(adapter.dispatch('merge_sql', 'elementary')(target_relation, tmp_relation, unique_key, dest_columns)) }}
+{% macro merge_sql(target_relation, tmp_relation, unique_key, dest_columns, incremental_predicates=none) -%}
+    {{ return(adapter.dispatch('merge_sql', 'elementary')(target_relation, tmp_relation, unique_key, dest_columns, incremental_predicates)) }}
 {%- endmacro %}
 
 {# Snowflake, Bigquery and Databricks #}
-{% macro default__merge_sql(target_relation, tmp_relation, unique_key, dest_columns) %}
-    {%- set merge_sql = dbt.get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns) %}
-    {{ return(merge_sql) }}
+{% macro default__merge_sql(target_relation, tmp_relation, unique_key, dest_columns, incremental_predicates) %}
+    {% set macro = dbt.get_merge_sql %}
+    {% if "incremental_predicates" in macro.get_macro().arguments %}
+      {% do return(macro(target_relation, tmp_relation, unique_key, dest_columns, incremental_predicates)) %}
+    {% endif %}
+    {% do return(macro(target_relation, tmp_relation, unique_key, dest_columns)) %}
```

Comment:
I'm guessing we can't do normal python stuff like `macro(*args, **kwargs)` right?